frame/composable-support/src/validation.rs

```diff
@@ -0,0 +1,98 @@
+use core::marker::PhantomData;
+use scale_info::TypeInfo;
+
+/// Black box that embbed the validated value.
+#[derive(Eq, PartialEq, Debug, TypeInfo, codec::Encode)]
+pub struct Validated<T, U> {
+	pub value: T,
+	_marker: PhantomData<U>,
+}
```

Comment:
Should also implement AsRef, Into, Etc. I'd suggest removing the pub modifier, so that constructing an invalid Validated becomes impossible.

Reply:
done